.github/workflows/versioning.yaml

```diff
@@ -54,10 +54,10 @@ jobs:
             );
             const changelog = readFileSync("pr/CHANGELOG.md")
               .toString()
-              .split(/^##?#?#?/m)
+              .split(/^## /m)
```

Comment:
I do hate having to write javascript in this indented, plain-text string way. I have thought about doing this, and in fact any time I make a change to these scripts I paste them into a separate temp `.js` file to format and check for errors, then paste back.

However as per my previous comment, I have a fervent preference for minimizing number of files in this template as much as I can. Some things I am forced to split into very fine granularity -- where otherwise I would avoid it -- for various reasons: `_styles` so users can find and work with them more easily, `/workflows` due to [GitHub limitations with chaining and reusing workflows](https://github.com/greenelab/lab-website-template/issues/153), `/_plugins` due to Jekyll/Liquid finicky-ness, etc. Breaking these out into separate scripts would be inconsistent with the level of granularity and design decision logic of the rest of the template. Also, the hardship of this only really affects the developers, not the users, so I am more than fine having that burden if it means the users wont have 5 extra files they need to manage, which is especially significant when the user is updating their template version and resolving merge conflicts.

Reply:
Thanks, I see what you mean here and that makes sense! I wonder if there are any extensions within IDE's which enable multi-lingual development within the contexts of yaml and other files (I don't know of any but maybe this is something for the wishlist!).